chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,151 @@
+use near_metrics::{try_create_gauge_vec, try_create_int_gauge};
+use near_store::db::{StatsValue, StoreStatistics};
+use once_cell::sync::Lazy;
+use prometheus::{GaugeVec, IntGauge};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::Mutex;
+use tracing::warn;
+
+pub(crate) fn export_stats_as_metrics(stats: StoreStatistics) {
+    match ROCKSDB_METRICS.lock().unwrap().export_stats_as_metrics(stats) {
+        Ok(_) => {}
+        Err(err) => {
+            warn!(target:"stats", "Failed to export store statistics: {:?}",err);
+        }
+    }
+}
+
+/// Wrapper for re-exporting RocksDB stats into Prometheus metrics.
+pub(crate) static ROCKSDB_METRICS: Lazy<Mutex<RocksDBMetrics>> =
+    Lazy::new(|| Mutex::new(RocksDBMetrics::default()));
+
+#[derive(Default, Debug)]
+/// Creates prometheus metrics on-demand for exporting RocksDB statistics.
+pub(crate) struct RocksDBMetrics {
+    // Contains counters and sums, which are integer statistics in RocksDB.
+    int_gauges: HashMap<String, IntGauge>,
+    // Contains floating point statistics, such as quantiles of timings.
+    gauges: HashMap<String, GaugeVec>,
+}
+
+impl RocksDBMetrics {
+    pub fn export_stats_as_metrics(
+        &mut self,
+        stats: StoreStatistics,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        for (stat_name, values) in stats.data {
+            if values.len() == 1 {
+                // A counter stats.
+                // A statistic 'a.b.c' creates the following prometheus metric:
+                // - near_a_b_c
+                if let StatsValue::Count(value) = values[0] {
+                    self.set_int_value(
+                        |stat_name: &str| stat_name.to_string(),
+                        |stat_name| get_prometheus_metric_name(stat_name),
+                        &stat_name,
+                        value,
+                    )?;
+                }
+            } else {
+                // A summary stats.
+                // A statistic 'a.b.c' creates the following prometheus metrics:
+                // - near_a_b_c_sum
+                // - near_a_b_c_count
+                // - near_a_b_c{quantile="0.95"}
+                for stats_value in values {
+                    match stats_value {
+                        StatsValue::Count(value) => {
+                            self.set_int_value(
+                                |stat_name| get_stats_summary_count_key(stat_name),
```

Comment:
```suggestion
                                get_stats_summary_count_key,
```


Just this might work here and elsewher